commit 112e68e5134e29667a57d9ee1a3799c0dafd45c6
Author: Ben Ashbaugh <ben.ashbaugh@intel.com>
Date:   Tue Nov 14 09:26:08 2023 -0800

    updates extension headers for the latest XML file (#240)
    
    Adds bits for cl_khr_initialize_memory
    Adds separate device handle lists for external memory and semaphores
    Adds enums for cl_arm_scheduling_controls

diff --git a/CL/cl_ext.h b/CL/cl_ext.h
index 3eba7ed..95c0a29 100644
--- a/CL/cl_ext.h
+++ b/CL/cl_ext.h
@@ -650,12 +650,18 @@ typedef cl_program
  * cl_khr_initialize_memory extension *
  **************************************/
 
-#define CL_CONTEXT_MEMORY_INITIALIZE_KHR            0x2030
+typedef cl_bitfield         cl_context_memory_initialize_khr;
 
+/* cl_context_properties */
+#define CL_CONTEXT_MEMORY_INITIALIZE_KHR                    0x2030
 
-/**************************************
- * cl_khr_terminate_context extension *
- **************************************/
+/* cl_context_memory_initialize_khr */
+#define CL_CONTEXT_MEMORY_INITIALIZE_LOCAL_KHR              (1 << 0)
+#define CL_CONTEXT_MEMORY_INITIALIZE_PRIVATE_KHR            (1 << 1)
+
+/***************************************************************
+* cl_khr_terminate_context
+***************************************************************/
 
 #define CL_CONTEXT_TERMINATED_KHR                   -1121
 
@@ -1262,8 +1268,8 @@ typedef cl_uint             cl_external_memory_handle_type_khr;
 #define CL_DEVICE_EXTERNAL_MEMORY_IMPORT_HANDLE_TYPES_KHR   0x204F
 
 /* cl_mem_properties */
-#define CL_DEVICE_HANDLE_LIST_KHR                           0x2051
-#define CL_DEVICE_HANDLE_LIST_END_KHR                       0
+#define CL_MEM_DEVICE_HANDLE_LIST_KHR                       0x2051
+#define CL_MEM_DEVICE_HANDLE_LIST_END_KHR                   0
 
 /* cl_command_type */
 #define CL_COMMAND_ACQUIRE_EXTERNAL_MEM_OBJECTS_KHR         0x2047
@@ -1442,8 +1448,8 @@ typedef cl_ulong            cl_semaphore_payload_khr;
 
 /* cl_semaphore_info_khr or cl_semaphore_properties_khr */
 #define CL_SEMAPHORE_TYPE_KHR                               0x203D
-/* enum CL_DEVICE_HANDLE_LIST_KHR */
-/* enum CL_DEVICE_HANDLE_LIST_END_KHR */
+#define CL_SEMAPHORE_DEVICE_HANDLE_LIST_KHR                 0x2053
+#define CL_SEMAPHORE_DEVICE_HANDLE_LIST_END_KHR             0
 
 /* cl_command_type */
 #define CL_COMMAND_SEMAPHORE_WAIT_KHR                       0x2042
@@ -1746,6 +1752,7 @@ typedef cl_bitfield cl_device_scheduling_controls_capabilities_arm;
 #define CL_DEVICE_SCHEDULING_REGISTER_ALLOCATION_ARM           (1 << 4)
 #define CL_DEVICE_SCHEDULING_WARP_THROTTLING_ARM               (1 << 5)
 #define CL_DEVICE_SCHEDULING_COMPUTE_UNIT_BATCH_QUEUE_SIZE_ARM (1 << 6)
+#define CL_DEVICE_SCHEDULING_COMPUTE_UNIT_LIMIT_ARM         (1 << 7)
 
 #define CL_DEVICE_SUPPORTED_REGISTER_ALLOCATIONS_ARM            0x41EB
 #define CL_DEVICE_MAX_WARP_COUNT_ARM                            0x41EA
@@ -1760,8 +1767,9 @@ typedef cl_bitfield cl_device_scheduling_controls_capabilities_arm;
 #define CL_KERNEL_EXEC_INFO_COMPUTE_UNIT_MAX_QUEUED_BATCHES_ARM 0x41F1
 
 /* cl_queue_properties */
-#define CL_QUEUE_KERNEL_BATCHING_ARM                            0x41E7
-#define CL_QUEUE_DEFERRED_FLUSH_ARM                             0x41EC
+#define CL_QUEUE_KERNEL_BATCHING_ARM                        0x41E7
+#define CL_QUEUE_DEFERRED_FLUSH_ARM                         0x41EC
+#define CL_QUEUE_COMPUTE_UNIT_LIMIT_ARM                     0x41F3
 
 /**************************************
 * cl_arm_controlled_kernel_termination
